src/Compilers/CSharp/Test/Semantic/Semantics/ColorColorTests.cs

```diff
@@ -1729,6 +1730,35 @@ class X
             CompileAndVerify(comp, expectedOutput: "42");
         }
 
+        [WorkItem(5362, "https://github.com/dotnet/roslyn/issues/5362")]
+        [Fact]
+        public void TestColorColorSymbolInfoInArrowExpressionClauseSyntax()
+        {
+            const string source = @"public enum Lifetime
+{
+    Persistent,
+    Transient,
+    Scoped
+}
+public class Example
+{
+    public Lifetime Lifetime => Lifetime.Persistent;
+    //                          ^^^^^^^^
+}";
+            var comp = CreateCompilationWithMscorlibAndSystemCore(source, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var syntaxRoot = syntaxTree.GetRoot();
+
+            var semanticModel = comp.GetSemanticModel(syntaxTree, false);
+
+            var actualSymbol = semanticModel.GetSymbolInfo(syntaxRoot.FindNode(TextSpan.FromBounds(130, 138)));
+            var expectedSymbol = semanticModel.GetDeclaredSymbol(syntaxRoot.FindNode(TextSpan.FromBounds(0, 71)));
+
+            Assert.Equal(expectedSymbol, actualSymbol.Symbol);
```

Comment:
We usually assert what the symbol is, so that the test would be more readable and easier to maintain. Usually this is done with Assert.Equal("...", expectedSymbol.ToTestDisplayString()).